core/src/main/com/sinch/sdk/core/databind/multipart/ObjectMapper.java

```diff
@@ -0,0 +1,127 @@
+package com.sinch.sdk.core.databind.multipart;
+
+import com.sinch.sdk.core.databind.FormSerializer;
+import com.sinch.sdk.core.databind.annotation.FormSerialize;
+import com.sinch.sdk.core.databind.annotation.PropertiesOrder;
+import com.sinch.sdk.core.databind.annotation.Property;
+import com.sinch.sdk.core.exceptions.SerializationException;
+import com.sinch.sdk.core.models.OptionalValue;
+import com.sinch.sdk.core.utils.EnumDynamic;
+import com.sinch.sdk.core.utils.Pair;
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.MethodDescriptor;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class ObjectMapper {
+
+  public Map<String, Object> serialize(Object value)
+      throws IntrospectionException,
+          InvocationTargetException,
+          IllegalAccessException,
+          ClassNotFoundException,
+          NoSuchMethodException,
+          InstantiationException {
+
+    BeanInfo beanInfo = Introspector.getBeanInfo(value.getClass(), Object.class);
+    List<Pair<String, Method>> serializableProperties = collectSerializableProperties(beanInfo);
+    return serializeProperties(serializableProperties, value);
+  }
+
+  private List<Pair<String, Method>> collectSerializableProperties(BeanInfo beanInfo) {
+
+    ArrayList<Pair<String, Method>> result = new ArrayList<>();
+    final MethodDescriptor[] methodDescriptors = beanInfo.getMethodDescriptors();
+
+    for (MethodDescriptor methodDescriptor : methodDescriptors) {
+      getPropertyGetter(methodDescriptor.getMethod()).ifPresent(result::add);
+    }
+    return sortProperties(beanInfo, result);
+  }
+
+  private List<Pair<String, Method>> sortProperties(
+      BeanInfo beanInfo, List<Pair<String, Method>> properties) {
+
+    PropertiesOrder propertyOrder =
+        beanInfo.getBeanDescriptor().getBeanClass().getAnnotation(PropertiesOrder.class);
+
+    if (null == propertyOrder) {
+      return properties;
+    }
+
+    ArrayList<Pair<String, Method>> sorted = new ArrayList<>(properties.size());
+
+    for (String property : propertyOrder.value()) {
+      properties.stream()
+          .filter(p -> p.getLeft().equals(property))
+          .findFirst()
+          .ifPresent(sorted::add);
```

Comment:
We can be more efficient here: you are repeatedly filtering through the properties list (O(n) for each property in propertyOrder). Instead, you can create a Map<String, Method> for faster lookups (O(1))

```suggestion
    // Convert the properties list to a map for efficient lookups
    Map<String, Method> propertyMap = properties.stream()
            .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));

    List<Pair<String, Method>> sorted = Arrays.stream(propertyOrder.value())
            .map(property -> Pair.of(property, propertyMap.get(property)))
            .filter(pair -> pair.getRight() != null)
            .collect(Collectors.toList());
```

And I'm not sure to understand how you handle the additional properties, but you then can add the following code to add them at the end of the sorted list:
```java
properties.stream()
    .filter(p -> !sorted.stream().anyMatch(s -> s.getLeft().equals(p.getLeft())))
    .forEach(sorted::add);
```

Reply:
Thank you to highlight the optimization !
Choose this implementation in place of you suggestion: https://github.com/sinch/sinch-sdk-java/pull/171/commits/b2d7acb077439ed9a113463befb0b5de4e880a75
It avoid to:
- create a new stream to create a map the collected to a list -> No need for 
```
    Map<String, Method> propertyMap = properties.stream()
            .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
```
- create a stream then Pairs to be collected to a List -> No need for 
```
    List<Pair<String, Method>> sorted = Arrays.stream(propertyOrder.value())
            .map(property -> Pair.of(property, propertyMap.get(property)))
            .filter(pair -> pair.getRight() != null)
            .collect(Collectors.toList());
```

Finally; only have:
```
    List<String> order = java.util.Arrays.asList(propertiesOrder.value());
    properties.sort(Comparator.comparingInt(l -> order.indexOf(l.getLeft())));
```
Still having to create a List from Array to have `order` variable
But the sort operation is performed directly onto the list received as parameter

Reply:
No more sorting, no more processing ✅